controllers/backplaneconfig_controller.go

```diff
@@ -312,11 +322,62 @@ func (r *MultiClusterEngineReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return result, err
 	}
 
+	if upgrade {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
 	r.StatusManager.AddCondition(status.NewCondition(backplanev1.MultiClusterEngineProgressing, metav1.ConditionTrue, status.DeploySuccessReason, "All components deployed"))
 
 	return ctrl.Result{}, nil
 }
 
+// This function set the operator condition created by OLM to either allow or disallow upgrade based on whether X.Y desired version matches current version
+// It returns an error as well as Boolean determining whether or not the reconcile needs to be rerun in order to update status
+
+func (r *MultiClusterEngineReconciler) setOperatorUpgradeableStatus(ctx context.Context, m *backplanev1.MultiClusterEngine) (bool, error) {
+
+	// Temporary variable
+	var upgradeable bool
+
+	// Checking to see if the current version of the MCE matches the desired to determine if we are in an upgrade scenario
+	// If the current version doesn't exist, we are currently in a install which will also not allow it to upgrade
+	parts1 := strings.Split(m.Status.CurrentVersion, ".")
+	parts2 := strings.Split(version.Version, ".")
+
+	if parts1[0] == parts2[0] && parts1[1] == parts2[1] {
+		upgradeable = true
+	} else {
+		upgradeable = false
+	}
+
+	// 	These messages are drawn from operator condition
+	// Right now, they just indicate between upgrading and not
+	msg := utils.UpgradeableAllowMessage
+	status := metav1.ConditionTrue
+	reason := utils.UpgradeableAllowReason
+
+	// 	The condition is the only field that affects whether or not we can upgrade
+	// The rest are just status info
+	if !upgradeable {
+		status = metav1.ConditionFalse
+		reason = utils.UpgradeableUpgradingReason
+		msg = utils.UpgradeableUpgradingMessage
```

Comment:
In the operatorCondition when I ran this it always showed
> message: 'upgrading the mce operator to version '
with a space at the end and no actual version number.